src/main/protobuf/spawn.proto

```diff
@@ -252,30 +263,93 @@ message ExecLogEntry {
   }
 
   // A set of spawn inputs.
-  // The contents of the set are the directly referenced files, directories and
-  // symlinks in addition to the contents of all transitively referenced sets.
+  // The contents of the set are the directly contained entries in addition to
+  // the contents of all transitively referenced sets. When order matters,
+  // transitive sets come before direct entries and within a set, entries are
+  // considered in left-to-right order ("postorder").
   // Sets are not canonical: two sets with different structure may yield the
   // same contents.
   message InputSet {
-    // Entry IDs of files belonging to this set.
-    repeated int32 file_ids = 1;
-    // Entry IDs of directories belonging to this set.
-    repeated int32 directory_ids = 2;
-    // Entry IDs of unresolved symlinks belonging to this set.
-    repeated int32 unresolved_symlink_ids = 3;
-    // Entry IDs of other sets contained in this set.
-    repeated int32 transitive_set_ids = 4;
+    // Entry IDs of files, directories, unresolved symlinks or runfiles trees
+    // belonging to this set.
+    repeated uint32 input_ids = 1;
+    reserved 2, 3;
```

Comment:
While implementing a consumer I noticed how painful it would be to support both versions if we were to reassign these IDs. I would thus lean towards reserving them. Do you consider that okay as well?

Reply:
For format detection, would it be better use the presence of one of the newly added fields in the `Invocation` message?

I thought reassigning might be better than reserving because, if you were to consume the new format using an old tool, reserving means depsets are silently (mis-)interpreted as empty, while reassigning makes it likely that the tool would crash, as some ids won't point to objects of the expected type. But I don't *super* care either way.

Reply:
> I suspect what you might be hinting at is that, with reserved fields, you can take a modified version of the proto with the removed fields added back, and use it to implement a single tool that understands both formats? But reassigning would require such a tool to interpret the proto contents at the byte level?

Yes, that's my point. I don't expect many external tools to consume the new format yet, but many users of such tools will be on Bazel < 7.4.0 for a while. If we want to make it easy for them to adopt the tools, I think that we should make it easy for tool authors to seamlessly support both.

Reply:
Understood, let's go with the current state.